pkg/controller/machine/controller.go

```diff
@@ -434,7 +435,16 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 		}
 
 		// Since we may have mutated the local copy of the machine above, we need to calculate baseToPatch here.
+		// Any updates to the status must be done after this point.
 		baseToPatch := client.MergeFrom(machine.DeepCopy())
+
+		if phase == phaseFailed {
+			if err := r.patchFailedMachineProviderStatusState(machine); err != nil {
```

Comment:
this is called `patchFailedMachineProviderStatusState` but is not actually patching but just setting right?
We should be consistent with a `patchFailedMachineInstanceAnnotation` and either let `patchFailedMachineProviderStatusState` to create a baseToPatch and do the patch requests or rename both funcs  and let them just set.

Reply:
I agree the naming is wrong, but I think the Instance annotation needs to stay as a patching function otherwise we have the issue of not actually updating the resource. The annotation needs to send a `client.Patch` but the rest of the changes need a `client.Status().Patch` because of the subresource.

I think the preferable route here is to just rename the new one not to include the patching word and leave the logic as is